config-common/src/super_config.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+// TODO: Move this to aziotctl-common
+
+/// This is the config stored in `/etc/aziot/config.toml`
+///
+/// It is an amalgam of the individual services' configs, with some tweaks:
+///
+/// - Wherever key IDs were used to link keys between services, the key is referenced directly.
+///   For example, instead of IS provisioning referencing a symmetric key by its key ID and KS optionally associating a key ID with a key URI,
+///   the IS provisioning optionally references the key URI directly.
+///
+/// - Wherever cert IDs were used to link certs between services, the key is referenced directly.
+///   For example, instead of IS provisioning referencing a cert by its cert ID and CS associating a cert ID with a cert URI or an issuance method,
+///   the IS provisioning references the cert URI or issuance method directly.
+///
+/// - A consequence of the above is that the EST endpoint spec only allows one of identity or bootstrap identity X.509 auth to be specified, not both.
+///   (If bootstrap identity was specified, the regular identity would be IDs.)
+///
+/// Unfortunately it's not easy to do this without duplicating all the individual services' config types.
+/// Inner types are reused when possible, ie when they don't need modification to work with the super-config.
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
```

Comment:
similarly, for consistency's sake, we've been using `use serde::{Deserialize, Serialize};` everywhere else in `aziotctl`

Reply:
Done.